src/screens/main/main.gd

```diff
@@ -2,20 +2,27 @@ extends Node2D
 
 const FIRST_SCN = "res://levels/procedural_test/levelgen.tscn"
 const MENU_SCN = ""
+var load_state = 0
+var cur_screen
 
 func _ready():
 	game.main = self
 	game.gui = $hud_layer/GameHUD
 	load_screen(FIRST_SCN, "", false)
 
-var load_state = 0
-var cur_screen
+func subscribe_to_player():
```

Comment:
I hoisted out the connections to the puppet master, now the sub can just emit its data and let subscriptions handle it.